client/src/components/PostForm.tsx

```diff
@@ -40,19 +41,27 @@ export const PostForm = (props: {
 }) => {
   const IS_EDITING = props.post !== undefined;
   const AuthToken = localStorage.getItem("BEARER_TOKEN") || "";
-
+  const photoArray = ((props.post?.postPhotos.length || 0) < 1 ? [] : props.post?.postPhotos) || [];
+  const [inputFields, setInputFields] = useState(
+    photoArray.map((photo) => {
+      return { imageUrl: photo.url };
+    })
+  );
   const [formState, setFormState] = useState({
     title: props.post?.title || "",
     content: props.post?.content || "",
     rating: props.post?.rating || 1,
-    imageUrl: props.post?.postPhotos[0]?.url || "",
+    imageUrl: props.post?.postPhotos || "",
   });
   const navigate = useNavigate();
   const toast = useToast();
 
   const createPost = async () => {
     if (!AuthToken) return navigate("/login");
     if (formState.title === "" || formState.content === "") return;
+    const imageUrlList = inputFields?.map((inputField) => {
+      return { url: inputField["imageUrl"] };
+    });
```

Comment:
Not too sure why theres a need for `inputFields` state and all the different states, just do everything on the `formState`.

```jsx
const [formState, setFormState] = useState({
    title: props.post?.title || "",
    content: props.post?.content || "",
    rating: props.post?.rating || 1,
    postPhotos: photoArray,   
  });
```
When adding a new image just simply add `{url: ""}` to the `postPhotos` array and send everything up to the API.